framework/src/org/checkerframework/common/signatureinference/SignatureInferenceScenes.java

```diff
@@ -0,0 +1,726 @@
+        }
+    }
+
+    // TODO: The two conversion methods below could be somewhere else. Maybe in AFU?
+    /**
+     * Converts an {@link javax.lang.model.element.AnnotationMirror}
+     * into an {@link annotations.Annotation}.
+     */
+    private static Annotation annotationMirrorToAnnotation(AnnotationMirror am) {
+        AnnotationDef def = new AnnotationDef(AnnotationUtils.annotationName(am));
+        Map<String, AnnotationFieldType> fieldTypes = new HashMap<>();
+        // Handling cases where there are fields in annotations.
+        for (ExecutableElement ee : am.getElementValues().keySet()) {
+            AnnotationFieldType aft = getAnnotationFieldType(ee, am.
+                    getElementValues().get(ee).getValue());
+            if (aft == null) return null;
+            // Here we just add the type of the field into fieldTypes.
+            fieldTypes.put(ee.getSimpleName().toString(), aft);
+        }
+        def.setFieldTypes(fieldTypes);
+
+        // Now, we handle the values of those types below
+        Map<? extends ExecutableElement, ? extends AnnotationValue> values =
+                am.getElementValues();
+        Map<String, Object> newValues = new HashMap<>();
+        for (ExecutableElement ee : values.keySet()) {
+            Object value = values.get(ee).getValue();
+            if (value instanceof List) {
+                @SuppressWarnings("unchecked")
+                List<Object> valueList = (List<Object>)value;
+                List<Object> newList = new ArrayList<>();
+                // If we have a List here, then it is a List of AnnotatedValue.
+                // Converting each AnnotatedValue to its respective Java type:
+                for (Object o : valueList) {
+                    newList.add(((AnnotationValue)o).getValue());
+                }
+                value = newList;
+            }
+            newValues.put(ee.getSimpleName().toString(), value);
+        }
+        Annotation out = new Annotation(def, newValues);
+        return out;
+    }
+
+    /**
+     * Converts an {@link annotations.Annotation} into an
+     * {@link javax.lang.model.element.AnnotationMirror}.
+     */
+    private static AnnotationMirror annotationToAnnotationMirror(
+            Annotation anno, ProcessingEnvironment processingEnv) {
+        final AnnotationBuilder builder = new AnnotationBuilder(processingEnv,
+                anno.def().name);
+        for (String fieldKey : anno.fieldValues.keySet()) {
+            addFieldToAnnotationBuilder(fieldKey,
+                    anno.fieldValues.get(fieldKey), builder);
+        }
+        return builder.build();
+    }
+
+    /**
+     * Returns an AnnotationFieldType given an ExecutableElement or value.
+     */
+    private static AnnotationFieldType getAnnotationFieldType(ExecutableElement
+            ee, Object value) {
+        if (value instanceof List<?>) {
+            // Handling cases of empty arrays was a bit troublesome here.
```

Comment:
What should a reader learn from this comment?  Is it of purely historical interest?  Or is there something that readers should keep in mind when looking at this code?